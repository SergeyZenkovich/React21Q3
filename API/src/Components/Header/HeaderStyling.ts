import React from "react";
import styled from "styled-components";
import fontPBold from "../../assets/fonts/MavenPro-Bold.ttf";

import fontP from "../../assets/fonts/MavenPro-Regular.ttf";

export const StyledHeaderBlock = styled.header`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 70px;
  background-color: #000a;
  margin-bottom: 10px;
  padding: 0;
`;
export const StyledNavigation = styled.nav`
  display: flex;
  justify-content: center;
  padding: 25px 0;
  & a {
    text-decoration: none;
  }
`;
export const StyledHeaderElement = styled.div`
  @font-face {
    font-family: "CardTextFontBold";
    src: url(${fontPBold});
  }
  width: 100px;
  margin: 0 10px;
  color: #fff;
  font-family: "CardTextFontBold";
  text-decoration: ${({activePage}: {activePage: boolean}) => (activePage ? "underline" : "none")};
`;
