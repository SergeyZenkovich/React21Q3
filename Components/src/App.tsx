import React from "react";
import styled from "styled-components";
import Card from "./Components/Card/Card";
import Serach from "./Components/SearchBar/SearchBar";
import cardsData from "./Data";
import {CardInterface} from "./interfaces/ComponentsInterfaces";

const StyledCardsBlock = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const App = (): JSX.Element => {
  const Cards = cardsData.map((el: CardInterface) => (
    <Card
      key={el.cardName}
      cardName={el.cardName}
      cardPrice={el.cardPrice}
      cardYear={el.cardYear}
      cardLikes={el.cardLikes}
      cardRating={el.cardRating}
      cardImg={el.cardImg}
    />
  ));
  return (
    <div>
      <Serach />
      <StyledCardsBlock>{Cards}</StyledCardsBlock>
    </div>
  );
};
export default App;
