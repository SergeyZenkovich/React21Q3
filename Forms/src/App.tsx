import React, {useState} from "react";
import styled from "styled-components";
import Card from "./Components/Card/Card";
import Form from "./Components/Form/Form";
import Serach from "./Components/SearchBar/SearchBar";
import cardsData from "./Data";
import {CardInterface} from "./interfaces/ComponentsInterfaces";

const StyledCardsBlock = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const App = (): JSX.Element => {
  const [cards, setCards] = useState([] as CardInterface[]);
  const addCard = (newCard: CardInterface): void => {
    setCards([...cards, newCard]);
  };
  const Cards = cards.map((el: CardInterface) => (
    <Card
      key={el.cardProblem}
      cardProblem={el.cardProblem}
      cardDate={el.cardDate}
      cardPriority={el.cardPriority}
      cardisRequestNeeded={el.cardisRequestNeeded}
    />
  ));
  return (
    <div>
      <Form addCard={addCard} />
      <StyledCardsBlock>{Cards}</StyledCardsBlock>
    </div>
  );
};
export default App;
