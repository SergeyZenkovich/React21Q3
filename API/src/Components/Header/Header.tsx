import React from "react";
import styled from "styled-components";

import {StyledHeaderBlock, StyledNavigation, StyledHeaderElement} from "./HeaderStyling";
import {NavElement} from "../../interfaces/ComponentsInterfaces";
import pagesData from "../../Data";

const Header = ({
  activePage,
  setActivePage
}: {
  activePage: string;
  setActivePage: React.Dispatch<React.SetStateAction<string>>;
}): JSX.Element => {
  const NavElements = pagesData.map((el: NavElement) => (
    <StyledHeaderElement key={el.page}>{el.page}</StyledHeaderElement>
  ));
  return (
    <StyledHeaderBlock>
      <StyledNavigation>{NavElements}</StyledNavigation>
    </StyledHeaderBlock>
  );
};

export default Header;
