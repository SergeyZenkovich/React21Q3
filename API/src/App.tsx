import React, {useEffect, useState} from "react";
import {Switch, Route} from "react-router-dom";
import styled from "styled-components";

import HomePage from "./Components/HomePage/HomePage";
import AboutPage from "./Components/AboutPage/AboutPage";
import Header from "./Components/Header/Header";
import ErrorBlock from "./Components/ErrorComponent/ErrorComponent";
import DetailsPage from "./Components/DetailsPage/DetailsPage";
import {FetchObject} from "./interfaces/ComponentsInterfaces";

const MainComponent = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;

const App = (): JSX.Element => {
  const [activePage, setActivePage] = useState("Home");
  const [pageElements, setPageElements] = useState([] as FetchObject[]);

  return (
    <div>
      <Header activePage={activePage} setActivePage={setActivePage} />
      <MainComponent>
        <Switch>
          <Route exact path="/about" render={() => <AboutPage setActivePage={setActivePage} />} />
          <Route
            exact
            path="/details/:id"
            render={() => <DetailsPage setActivePage={setActivePage} />}
          />
          <Route
            exact
            path="/"
            render={() => (
              <HomePage setActivePage={setActivePage} setPageElements={setPageElements} />
            )}
          />
          <Route path="/*" render={() => <ErrorBlock />} />
        </Switch>
      </MainComponent>
    </div>
  );
};
export default App;
