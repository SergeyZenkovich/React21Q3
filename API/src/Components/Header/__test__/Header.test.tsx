import React, {useState} from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import styled from "styled-components";
import "jest-styled-components";
import {act} from "react-dom/test-utils";
import {Provider} from "react-redux";
import {HashRouter} from "react-router-dom";
import Header from "../Header";
import App from "../../../App";
import {StyledHeaderBlock, StyledHeaderElement} from "../HeaderStyling";
import store from "../../../redux/store";

const testActivePage = "Home";

it("renders without crashing", () => {
  const container = document.createElement("div");
  ReactDOM.render(
    <HashRouter>
      <Provider store={store}>
        <Header activePage={testActivePage} setActivePage={() => {}}></Header>
      </Provider>
    </HashRouter>,
    container
  );
  const innerBlockClassName = StyledHeaderBlock.styledComponentId;
  const innerBlock = container.querySelector(`.${innerBlockClassName}`);
  expect(container.contains(innerBlock)).toBeTruthy();
});

test("corectly render header elements with active styles", () => {
  const tree = renderer.create(<StyledHeaderElement activePage={true} />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("text-decoration", "underline");
});
test("corectly render header elements without active styles", () => {
  const tree = renderer.create(<StyledHeaderElement activePage={false} />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("text-decoration", "none");
});

describe("Header Component", () => {
  const setFunction = jest.fn();
  let container;
  beforeEach(() => {
    container = renderer.create(
      <HashRouter>
        <Provider store={store}>
          <Header activePage={testActivePage} setActivePage={setFunction}></Header>
        </Provider>
      </HashRouter>
    );
  });
  it("matches shapshot", () => {
    const tree = container.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
