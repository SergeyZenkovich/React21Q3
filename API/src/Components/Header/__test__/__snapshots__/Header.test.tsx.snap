// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches shapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 70px;
  background-color: #000a;
  margin-bottom: 10px;
  padding: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 25px 0;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  width: 100px;
  margin: 0 10px;
  color: #fff;
  font-family: "CardTextFontBold";
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  width: 100px;
  margin: 0 10px;
  color: #fff;
  font-family: "CardTextFontBold";
  -webkit-text-decoration: none;
  text-decoration: none;
}

<header
  className="c0"
>
  <nav
    className="c1"
  >
    <a
      aria-current="page"
      className="active"
      href="#/"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="c2"
      >
        Home
      </div>
    </a>
    <a
      aria-current={null}
      href="#/about"
      onClick={[Function]}
    >
      <div
        className="c3"
      >
        About
      </div>
    </a>
  </nav>
</header>
`;

exports[`corectly render header elements with active styles 1`] = `
.c0 {
  width: 100px;
  margin: 0 10px;
  color: #fff;
  font-family: "CardTextFontBold";
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="c0"
/>
`;

exports[`corectly render header elements without active styles 1`] = `
.c0 {
  width: 100px;
  margin: 0 10px;
  color: #fff;
  font-family: "CardTextFontBold";
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="c0"
/>
`;
